### ISSUE TITLE
Segfaults of CMake with libiorouter

### ISSUE THREAD

**Issue by fff7d1bc** [v92/libiorouter#26]
Hello,

I am trying to use libiorouter inside CI process and I am facing some issues, starting with CMake.

For the reproduce purpose I am using the default libiorouter.h with env LIBIOR_IO=on.

Steps to reproduce:

```
mkdir -p /tmp/cmake-test/build
cd /tmp/cmake-test/build
echo 'program(foo C CXX)' > ../CMakeLists.txt
LIBIOR_IO=on LD_PRELOAD=/path/to/libiorouter.so cmake ..
```

Strace output:

```
(...)
getcwd("/tmp/cmake-test/build", 2048)   = 22
lstat("/tmp", {st_mode=S_IFDIR|S_ISVTX|0777, st_size=140, ...}) = 0
lstat("/tmp/cmake-test", {st_mode=S_IFDIR|0700, st_size=80, ...}) = 0
lstat("/tmp/cmake-test/build", {st_mode=S_IFDIR|0700, st_size=60, ...}) = 0
getcwd("/tmp/cmake-test/build", 4096)   = 22
access("cmake", R_OK)                   = -1 ENOENT (No such file or directory)
access("/home/piotr/bin/cmake", R_OK)   = -1 ENOENT (No such file or directory)
access("/usr/local/bin/cmake", R_OK)    = -1 ENOENT (No such file or directory)
access("/usr/bin/cmake", R_OK)          = 0
stat("/usr/bin/cmake", {st_mode=S_IFREG|0755, st_size=4508920, ...}) = 0
access("/usr/bin/cmake", R_OK)          = 0
access("/usr/bin/cmake", R_OK)          = 0
lstat("/usr", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/usr/bin", {st_mode=S_IFDIR|0755, st_size=69632, ...}) = 0
lstat("/usr/bin/cmake", {st_mode=S_IFREG|0755, st_size=4508920, ...}) = 0
access("/usr/bin/cmake-gui", R_OK)      = -1 ENOENT (No such file or directory)
access("/usr/bin/ccmake", R_OK)         = 0
access("/usr/share/cmake/Modules/CMake.cmake", R_OK) = 0
getcwd("/tmp/cmake-test/build", 2048)   = 22
getcwd("/tmp/cmake-test/build", 4096)   = 22
stat("..", {st_mode=S_IFDIR|0700, st_size=80, ...}) = 0
getcwd("/tmp/cmake-test/build", 2048)   = 22
access("/tmp/cmake-test/CMakeCache.txt", R_OK) = -1 ENOENT (No such file or directory)
access("/tmp/cmake-test/CMakeLists.txt", R_OK) = 0
getcwd("/tmp/cmake-test/build", 2048)   = 22
access("/tmp/cmake-test/build/CMakeCache.txt", R_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/cmake-test/build/CMakeFiles/", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("", 0x7ffee756d3f0)                = -1 ENOENT (No such file or directory)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0} ---
+++ killed by SIGSEGV +++
```

**Comment by v92** [comment]
Hi,
bug that caused this was fixed in fa15c72626584b4817d3a3a0be47bf723f65558f . But I couldn't make it work so I had to turn off:
- directory handling 
- whiteout optimization 
  and now I was able to build nfs-ganesha which uses cmake. So try it.

**Comment by fff7d1bc** [comment]
No more segfaults, thanks.
